Include database error in the exception message.

dd32 opened the pull request:
In #966 the "Database error." exception was improved to include the actual DB error, however, it only updated one location where this exception is used.

This sync's the code to the other location.

100% untested.

dd32 committed:
Include database error in the exception message.

dd32 committed:
Include database error in the exception message.

barryhughes committed:
Add changelog entry.

barryhughes reviewed (commented):
Nice, low overhead and probably worth including for improved consistency with the `DBStore`.

Note that the `wpPostStore` mostly sits unused on modern installations (the current default is `DBStore`). A temporary exception to this crops up when a plugin is activated/deactivated and, though generally only for a very short period of time, it will then be invoked via the `HybridStore`.

So, to test, I forced the use of `wpPostStore` with:

```php
add_filter( 'action_scheduler_store_class', fn () => ActionScheduler_wpPostStore::class, 1000 );
```

I then deliberately introduced an error to the claim process using:

```php
add_filter( 'query', function ( $query ) {
	return class_exists( ActionScheduler_wpPostStore::class ) && str_contains( $query, ActionScheduler_wpPostStore::POST_TYPE )
		? $query . ' __SCUPPERED__'
		: $query;
} );
```

And triggered a queue run via:

```sh
wp action-scheduler run
```

The resulting database error was dutifully reported in the command output:

```
Error: There was an error running the action scheduler: Unable to claim actions. Database error: You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near '__SCUPPERED__' at line 1.
```

As probably true of the original work for the `DBStore`, this may anyway be caught and logged in various environments because `wpdb` makes its own `error_log()` call, but again it improves consistency and helps ensure the problem is surfaced *(perhaps* at the cost of adding noise, in some environments).

barryhughes committed:
Remove unnecessary `\` qualification of RuntimeException.

This shouldn't really matter, but it does matter to our current phpcs linting config (which otherwise can't seem to match the exception to the `@throws` tag).

barryhughes committed:
Pass output through escaping function.

Arguably this isn't the best place to do it, but the overhead is minimal and it satisfies our current linting rules.

barryhughes committed:
Rework escaping of exception message.

This is chiefly to satisfy the linter, is low overhead.
